### ISSUE TITLE
Nodes are not deleted from existing RKE2 cluster when clusters are deleted when RKE2 feature flag is disabled

### ISSUE THREAD

**Issue by sowmyav27** [rancher/rancher#35179]
**Rancher Server Setup**
- Rancher version: v2.6.1
- Installation option (Docker install/Helm Chart): docker install
- Proxy/Cert Details: rancher generated

**Information about the Cluster**
- Kubernetes version: `v1.21.5+rke2r2` 
- Cluster Type (Local/Downstream): RKE2 DO

**Describe the bug**
RKE2 nodes are not deleted from DO backend when clusters are deleted

**To Reproduce**
- Deployed 2 rke2 DO cluster
- **Disable RKE2 feature in rancher**
- Deploy an RKE1 cluster
- From Cluster Management, delete the clusters
- All the clusters are stuck in removing state and the RKE2 nodes are not deleted from the backend. 
- RKE1 cluster nodes seem to have been deleted from DO backend

**Comment by ronhorton** [comment]
**PASS** Verified in `2.6-head Commit ID 638b29a`

`The verification of this ticket has changed to ensuring that the user cannot disable the RKE2 flag once there are RKE2 clusters deployed`

1. Created 2 RKE2 DO clusters
2. Went to hamburger > global settings > feature flags
3. Disabled rke2 feature flag
4. rke2 flag remains active.

**Comment by ronhorton** [comment]
**Cannot deleted rke2 cluster**
1. created an rke2 cluster
2. attempted to deactivate rke2 feature flag - remains active
3. delete rke1 and rke2 clusters from cluster management page

result: `rke2 is not deleted`
<img width="989" alt="Screen Shot 2021-11-11 at 1 18 17 PM" src="https://user-images.githubusercontent.com/89212856/141351583-23d58c07-f14c-4190-ae82-995984f6a228.png">

**Comment by ronhorton** [comment]
**PASS** Verified in 2.6-head Commit ID 78b25c3

1. Created downstream RKE2 EC2 cluster
2. Created downstream RKE1 EC2 cluster
3. Created downstream RKE1 DO cluster
4. Created downstream RKE2 DO cluster
5. attempted to deactivate RKE2 Feature Flag Cannont deactivate RKE2 feature flag with existing RKE2 cluster(s)
6. Successfully deleted RKE2 clusters
7. Deactivated RKE2 feature flag flag was deactivated as there are no RKE2 clusters
8. Successfully deleted RKE1 clusters

`RKE1 and RKE2 Clusters were deleted from the UI and deleted from the backend.`

**Comment by ronhorton** [comment]
One use case remains for this ticket:
1. On 2.6.2 deploy  rke2 and rke1 clusters
2. Disable rke2 feature flag
3. Upgrade to 2.6head
4. Delete RKE2 cluster --> will this work?
5. Enable rke2 feature flag
6. Delete rke2 and rke1 clusters --> Will this work?

last use case is blocked by https://github.com/rancher/fleet/issues/607

**Comment by ronhorton** [comment]
Created new issue https://github.com/rancher/rancher/issues/35696 to address upgrade test.

1. On 2.6.2 deploy rke2 and rke1 clusters
2. Disable rke2 feature flag
3. Upgrade to 2.6head
4. Delete RKE2 cluster --> will this work?
5. Enable rke2 feature flag
6. Delete rke2 and rke1 clusters --> Will this work?

**Comment by ronhorton** [comment]
**PASS** Verified in `2.6-head Commit ID 4375161`
1. created rke2 do cluster
2. created rke1 do cluster
3. deleted rke2 cluster - removed from rancher, and deleted at back end
4. `disabled rke2 feature flag`
5. created 2nd rke1 do cluster
6. deleted both rke1 clusters

`All clusters were removed from rancher UI and deleted on backend`
